pkg/controller/apikey/index.go

```diff
@@ -18,43 +18,29 @@ import (
 	"net/http"
 
 	"github.com/google/exposure-notifications-verification-server/pkg/controller"
-	"github.com/google/exposure-notifications-verification-server/pkg/controller/flash"
 	"github.com/google/exposure-notifications-verification-server/pkg/database"
-	"github.com/gorilla/csrf"
 )
 
 func (c *Controller) HandleIndex() http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		ctx := r.Context()
-		flash := flash.FromContext(w, r)
-
-		user := controller.UserFromContext(ctx)
-		if user == nil {
-			flash.Error("Unauthorized.")
-			http.Redirect(w, r, "/signout", http.StatusSeeOther)
-			return
-		}
 
 		realm := controller.RealmFromContext(ctx)
 		if realm == nil {
-			flash.Error("Select a realm to continue.")
-			http.Redirect(w, r, "/realm", http.StatusSeeOther)
+			controller.MissingRealm(w, r, c.h)
 			return
 		}
 
 		// Perform the lazy load on authorized apps for the realm.
 		if _, err := realm.GetAuthorizedApps(c.db, true); err != nil {
-			flash.ErrorNow("Failed to load API Keys: %v", err)
+			controller.InternalError(w, r, c.h, err)
+			return
 		}
 
 		m := controller.TemplateMapFromContext(ctx)
-		m["user"] = user
-		m["realm"] = realm
 		m["apps"] = realm.AuthorizedApps
-		m["flash"] = flash
 		m["typeAdmin"] = database.APIUserTypeAdmin
 		m["typeDevice"] = database.APIUserTypeDevice
-		m[csrf.TemplateTag] = csrf.TemplateField(r)
```

Comment:
This is a good example of the enhancement - all these fields are automagically set by the middlewares now.

Reply:
yay